python/pyspark/pandas/internal.py

```diff
@@ -385,10 +528,10 @@ def __init__(
         spark_frame: spark.DataFrame,
         index_spark_columns: Optional[List[spark.Column]],
         index_names: Optional[List[Optional[Tuple]]] = None,
-        index_dtypes: Optional[List[Dtype]] = None,
+        index_fields: Optional[List[Field]] = None,
```

Comment:
Hm .. should we maybe combine it with `index_spark_columns` and `data_spark_columns` respectively?

Reply:
Can be done in a followup but just wondering.